src/builder.rs

```diff
@@ -883,17 +987,18 @@ impl<P: InProgressProofs, V> Bundle<InProgress<P, PartiallyAuthorized>, V> {
     pub fn sign<R: RngCore + CryptoRng>(self, mut rng: R, ask: &SpendAuthorizingKey) -> Self {
         let expected_ak = ask.into();
         let sighash = self.authorization().sigs.sighash;
-        self.map_authorization((
-            |proof| proof,
-            |proof| proof,
-            |maybe| match maybe {
-                MaybeSigned::SigningMetadata(parts) if parts.ak == expected_ak => {
-                    MaybeSigned::Signature(ask.randomize(&parts.alpha).sign(&mut rng, &sighash))
+        self.map_authorization(
+            &mut rng,
+            |_, proof| proof,
+            |_, proof| proof,
+            |rng, maybe| match maybe {
+                MaybeSigned::SigningParts(parts) if parts.ak == expected_ak => {
+                    MaybeSigned::Signature(ask.randomize(&parts.alpha).sign(rng, &sighash))
                 }
                 s => s,
             },
-            |partial| partial,
-        ))
+            |_rng, partial| partial,
```

Comment:
To match all the other occurrences in this file:
```suggestion
            |_, partial| partial,
```

Reply:
Ah, this was a `clippy --fix` change.